pkg/utils/utils.go

```diff
@@ -237,6 +238,49 @@ func convert(strategy esv1beta1.ExternalSecretConversionStrategy, str string) st
 	return strings.Join(newName, "")
 }
 
+// RevertKeys reverts a secret map into a valid key map as expected by push secrets.
+// Replaces encoded characters back to the original value depending on convert strategy.
+func RevertKeys(strategy esv1alpha1.PushSecretConversionStrategy, in map[string][]byte) (map[string][]byte, error) {
+	out := make(map[string][]byte, len(in))
+	for k, v := range in {
+		key := revert(strategy, k)
+		if _, exists := out[key]; exists {
+			return nil, fmt.Errorf("secret name collision during conversion: %s", key)
+		}
+		out[key] = v
+	}
+	return out, nil
+}
+
+func revert(strategy esv1alpha1.PushSecretConversionStrategy, str string) string {
+	switch strategy {
+	case esv1alpha1.PushSecretConversionReverseUnicode:
+		unicodeRegex := regexp.MustCompile(`_U([0-9a-fA-F]{4,5})_`)
```

Comment:
Declare this as a package variable so it doesn't get re-created every time this loop runs. That is a wasteful operation.

Reply:
thanks for the suggestion, I have moved it to a package variable.